src/core/internal/traits.d

```diff
@@ -164,6 +164,75 @@ template isInnerClass(T) if (is(T == class))
         enum isInnerClass = false;
 }
 
+/// Returns the type of `alias T` if it is not a type, or just returns `T`
+/// if it is a type
+private template Type(alias T)
+{
+    static if (is(typeof(T) == X, X))
+    {
+        alias Type = X;
+    }
+    else
+    {
+        alias Type = T;
+    }
+}
+
+/// Detect whether type `T` is a static array.
+template isStaticArray(T)
```

Comment:
Question:
With dlang/dmd#9769 template alias parameters should accept all types and hence this overload would not be required. But removing it results in the following error message:

```
traits.d(196): Error: template instance `isStaticArray!(int[3])` does not match template declaration isStaticArray(alias T)
traits.d(196):        while evaluating: static assert(isStaticArray!(int[3]))
```

Reply:
I don't know. @FeepingCreature ?

Reply:
Works with run.dlang.org on dmd-nightly... https://run.dlang.io/is/DYSMCo

Are you sure you're using current dmd?

Reply:
@FeepingCreature My bad, i was using 2.086.1.